### PR TITLE
Add getTypeAllocSize

### PR DESCRIPTION
**Pull request by vihanb**
Adds the `DataLayout.getTypeAllocSize()` to `DataLayout`, useful for things like `malloc()` etc.

### CODE REVIEW

**Comment by vihanb** [comment]
oh, oops, looks like this has things from the ConstantExpr PR, I'll remove that

File: llvm-node.d.ts [hunk 1]
```diff
@@ -173,6 +173,16 @@ declare namespace llvm {
     isAllOnesValue(): boolean;
   }
 
+  class ConstantExpr extends Constant {
```
Comment (MichaReiser) [inline llvm-node.d.ts]: It still seems like your ConstantExpr changes are in this PR, Maybe its best if you create a new branch only containing the getTypeAllocSize changes?

File: src/ir/integer-type.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+public:
+    static NAN_MODULE_INIT(Init);
+    static v8::Local<v8::Object> of(llvm::IntegerType *type);
+    using FromValueMixin<IntegerTypeWrapper>::FromValue;
+
+    llvm::IntegerType* getIntegerType();
+    static bool isInstance(v8::Local<v8::Value> value);
+
+protected:
+    llvm::Type* type;
```
Comment (MichaReiser) [inline src/ir/integer-type.h]: This should probably be an instance of IntegerType?

**Comment by vihanb** [comment]
What I think might be easier is I'll just perhaps put all my changes in one PR that way I don't have to figure out to remove some changes with git :P

**Comment by MichaReiser** [comment]
This works as well. 

Another option would either be to do an interactive-rebase from the master revision and skipping the undesired commits. Or you could also create a new branch, and cherry-pick the one commit you are interested in from this branch.

But do what's easiest for you. 👍

**Pull request closed without merging**